pkg/efivarfs/fs_test.go

```diff
@@ -20,32 +20,24 @@ func TestFSGoodFile(t *testing.T) {
 	}
 	i, err := getInodeFlags(f)
 	if err != nil {
-		t.Skipf("Can not getInodeFlags: %v != nil", err)
+		t.Skipf("Cannot getInodeFlags: %v != nil", err)
 	}
 
 	if err := setInodeFlags(f, i); err != nil {
 		t.Fatalf("setInodeFlags: %v != nil", err)
 	}
 
-	restore, err := makeMutable(f)
-	if err != nil {
-		t.Fatalf("makeMutable: %v != nil", err)
+	if err := setInodeFlags(f, i|unix.STATX_ATTR_IMMUTABLE); err != nil {
+		t.Skipf("Skipping rest of test, unable to set immutable flag: %+v", err)
```

Comment:
if fact it NEVER returns a nil restore function, hmm.

Reply:
It can return nil, but only if err is non-nil, so once we've handled the err != nil case we know we have a non-nil restore.

However, it does sometimes return a noop (`func() {}`) when the file was already mutable.